### PR TITLE
fix: support remote host in CDP discovery

### LINKED ISSUE #851: CDP Remote Optimization
The latest version 0.20.6 can only link to CDP remote browsers through tokens. 

 Attached are the test cases：
Since the browser disabled access to the real IP, a proxy was first set up to forward from port 9223 to 9222, and Chrome was launched
netsh interface portproxy add v4tov4 listenport=9223 listenaddress=0.0.0.0 connectport=9222 connectaddress=127.0.0.1    

visit  http://10.211.55.12:9223/json
image

When using http://10.211.55.12:9223 Or ws://10.211.55.12:9223 is invalid
image

Only by providing a WS token can access be granted.I hope to link directly through IP and port because the token will change every time we open the browser.
Please help.

### PR DESCRIPTION
**Pull request by jin-2-kakaoent**
## Summary

Fixes #851

- `discover_cdp_url` now accepts a `(host, port)` parameter instead of hardcoding `127.0.0.1`, allowing `connect "http://<remote-ip>:<port>"` to query the correct remote `/json/version` endpoint
- The returned `webSocketDebuggerUrl` is rewritten to match the requested host and port, since Chrome always reports `127.0.0.1` regardless of the interface it was reached through
- Extracted common `fetch_cdp_info` helper to eliminate duplicated HTTP fetch + JSON parsing logic between `discover_cdp_url` and `discover_cdp_url_with_timeout`

## Reproduction

Simulated a remote CDP browser on macOS by binding a TCP proxy to the LAN IP only:

```bash
# 1) Launch Chrome with debugging port (binds to 127.0.0.1 only)
chrome --remote-debugging-port=9222 --user-data-dir=/tmp/chrome-debug

# 2) Create a LAN-only TCP proxy (simulates remote browser)
# Binds to LAN IP only — 127.0.0.1:9224 is NOT reachable
python3 -c "
import socket, threading
def forward(src, dst):
    try:
        while True:
            data = src.recv(4096)
            if not data: break
            dst.sendall(data)
    except: pass
    finally: src.close(); dst.close()

server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
server.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
server.bind(('<LAN_IP>', 9224))  # LAN IP only
server.listen(5)
while True:
    client, _ = server.accept()
    remote = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    remote.connect(('127.0.0.1', 9222))
    threading.Thread(target=forward, args=(client, remote), daemon=True).start()
    threading.Thread(target=forward, args=(remote, client), daemon=True).start()
"
```

**Before fix:**
```
$ agent-browser connect "http://<LAN_IP>:9224"
✗ Failed to connect to CDP on port 9224: error sending request for url (http://127.0.0.1:9224/json/version)
```

**After fix:**
```
$ agent-browser connect "http://<LAN_IP>:9224"
✓ Done
```

## Test plan

- [x] Unit tests for `rewrite_ws_host` covering IPv4, IPv6, and edge cases
- [x] Manually reproduced the issue using a LAN-only TCP proxy and verified the fix resolves it

### CODE REVIEW

File: cli/src/native/cdp/discovery.rs [hunk 1]
```diff
@@ -2,26 +2,67 @@ use std::time::Duration;
...
+}
 
-    let body = tokio::time::timeout(request_timeout, async { reqwest_get_string(&url).await })
+/// Fetch `/json/version` from the given host:port and parse the response.
+async fn fetch_cdp_info(
+    host: &str,
+    port: u16,
+    timeout: Duration,
+) -> Result<BrowserVersionInfo, String> {
+    let url = format!("http://{}:{}/json/version", host, port);
```
[the remark this answers is not in the document]
Comment (hyunjinee) [reply cli/src/native/cdp/discovery.rs]: [70effd3](https://github.com/vercel-labs/agent-browser/pull/854/commits/70effd3ca86ed7f67b4bd363df7d0b59b38ffd4f)

**Comment by ctate** [comment]
Very nice. Thanks @jin-2-kakaoent!
